mav_trajectory_generation/src/polynomial.cpp

```diff
@@ -20,8 +20,113 @@
 
 #include "mav_trajectory_generation/polynomial.h"
 
+#include <algorithm>
+#include <limits>
+
 namespace mav_trajectory_generation {
 
+void Polynomial::findMinMaxCandidates(
+    double t_start, double t_end,
+    const Eigen::VectorXcd& roots_derivative_of_derivative,
+    std::vector<double>* candidates) const {
+  CHECK_NOTNULL(candidates);
+  candidates->clear();
+  candidates->reserve(roots_derivative_of_derivative.size() + 2);
+  candidates->push_back(t_start);
+  candidates->push_back(t_end);
+  for (size_t i = 0; i < roots_derivative_of_derivative.size(); i++) {
+    // Only real roots are considered as critical points.
+    if (std::abs(roots_derivative_of_derivative[i].imag()) >
+        std::numeric_limits<double>::epsilon()) {
+      continue;
+    }
+    const double candidate = roots_derivative_of_derivative[i].real();
+    // Do not evaluate points outside the domain.
+    if (candidate < std::min(t_start, t_end) ||
```

Comment:
I would hope t_start and t_end would be ordered correctly. Why not just enforce that instead?